week2/index.js

```diff
@@ -0,0 +1,163 @@
+            }
+            resolve(data);
+
+        })
+    })
+
+};
+
+let writeFiles = (filename, dataAsString) => {
+    return new Promise((resolve, reject) => {
+        fs.writeFile(TODO_FILE, dataAsString, (err) => {
+            if (err) {
+                if (err.code === "ENOENT") {
+                    // No such file or directory
+                    console.log("Sorry, this file doesn't exist.");
+                } else if (err.code === "EACCES") {
+                    // the user don't have a permission
+                    console.log("Sorry, but you don't have permission.");
+                } else {
+                    // any other Error
+                    console.log(` unknown error occurred: ${err.message}`);
+                }
+                process.exit();
+            }
+            resolve();
+        })
+    })
+}
+
+let addTodoItem = (filename, todoList, itemText) => {
+    todoList.push({
+        TASK_NUMBER: todoList.length + 1,
+        TODO_TASK: itemText
+    });
+    writeFiles(filename, JSON.stringify(myTodoList)).then(() => console.log("\nDone Adding the task on your TODO list.\n\n")).then(() => askingLoop());
+}
+
+let removeTodoItem = (filename, todoList, taskNumber) => {
+    todoList.splice([taskNumber - 1], 1);
+    for (let i = 0; i < todoList.length; i++) {
+        todoList[i].TASK_NUMBER = i + 1;
+    }
+    writeFiles(filename, JSON.stringify(myTodoList)).then(() => console.log("\nDone Removing the task from your TODO list.\n")).then(() => askingLoop());
+}
+
+let updateTodoItem = (filename, todoList, taskNumber, itemText) => {
+    todoList[taskNumber - 1] = {
+        TASK_NUMBER: taskNumber,
+        TODO_TASK: itemText
+    };
+    writeFiles(filename, JSON.stringify(myTodoList)).then(() => console.log("\nDone Updating the task on your TODO list.\n")).then(() => askingLoop());
+}
+
+let resetTodoItem = (filename, todoList) => {
+    todoList = [];
+}
+
+const readline = require('readline');
+const rl = readline.createInterface(process.stdin, process.stdout);
+
+function askingLoop(data) {
+
+    rl.question('Write your commend > ', function (data) {
+
+        switch (data.toString().trim()) {
+            case 'help':
+                readFiles(HELP_MENU).then((data) => {
+                    console.log(data);
+                    process.stdout.write('\nWrite your commend > ')
+                });
+
+                break;
+            case 'list':
+                console.log(`\n----------------------------------------------------------------------------------\n`);
+                myTodoList.map(task => { console.log(`    ${task.TASK_NUMBER}:${task.TODO_TASK}`) });
+                console.log(`\n----------------------------------------------------------------------------------\n`);
+                process.stdout.write('\nWrite your commend > ');
+                break;
+            case 'add':
+
+                rl.question('\nWrite a new task or "E" to exit> ', function (task) {

```

Comment:
What if my new task is called `"E"`? Seems like an unnecessary limitation, since I could use Ctrl-C to force cancel if I made a mistake.